src/components/contestdetails/ContestDetails.tsx

```diff
@@ -17,252 +17,182 @@ import AdminControls from "../admin/AdminControls";
 import { Contest, ContestStatus, Rank, Submission, Vote } from "../constants/Constants";
 import ContestSubmissionResults from "../ContestSubmissionResults";
 import OpenContestDetails from "./OpenContestDetails";
+import { useSnapshot } from "../../hooks";
 
 type ContestDetailsProps = {
   contestId: string // The contest to show details for
   onExit: () => void
 }
 
-type ContestDetailsState = {
-  contest?: Contest
-  showViewer: boolean
-  activeViewerSubmission?: Submission
-  submissions?: Submission[]
-  selectedRank?: Rank
-  votes?: Vote[]
-}
-
-export default class ContestDetails extends Component<ContestDetailsProps, ContestDetailsState> {
-  private userFolder: StorageReference 
-  private unsubscribeContest: Unsubscribe = () => { }
-  private unsubscribeSubmissions: Unsubscribe = () => { }
-  private unsubscribeVotes: Unsubscribe = () => { }
-
-  constructor(props: any) {
-    super(props)
-
-    this.state = { showViewer: false }
-    this.userFolder = ref(storage, auth.currentUser?.uid)
-
-    this.onRankClick = this.onRankClick.bind(this)
-    this.onClickSubmission = this.onClickSubmission.bind(this)
-    this.onResetVotesClick = this.onResetVotesClick.bind(this)
-    this.onSelectContestStatus = this.onSelectContestStatus.bind(this)
-  }
-
-  async componentDidMount(): Promise<void> {
-    this.unsubscribeContest = onSnapshot(
-      doc(db, 'contests', this.props.contestId),
-      (contest) => {
-        this.setState({ contest: { id: contest.id, ...contest.data()} as Contest })
-      }
-    )
-
-    this.unsubscribeSubmissions = onSnapshot(
-      collection(db, 'contests', this.props.contestId, 'submissions'),
-      (submissionDocs) => {
-        this.setState({
-          submissions: submissionDocs.docs.map(
-            (doc) => ({
-              submissionId: doc.id!,
-              userId: doc.id!,
-              ...doc.data()
-            } as Submission)
-          )
-        })
-      }
+export default function ContestDetails(props: ContestDetailsProps) {
+  const [showViewer, setShowViewer] = useState(false)
+  const [activeViewerSubmission, setactiveViewerSubmission] = useState<Submission>()
+  const [selectedRank, setSelectedRank] = useState<Rank>()
+  const contest = useSnapshot(doc(db, 'contests', props.contestId) as DocumentReference<Contest>)
+  const submissions = useSnapshot(collection(db, 'contests', props.contestId, 'submissions') as CollectionReference<Submission>)
+  const votes = useSnapshot(collection(db, 'contests', props.contestId, 'votes') as CollectionReference<Vote>)
+
+  function currentUserVotes(submissionId?: string) {
```

Comment:
This needs to be reverted, it changed from an "and" filter to an "or" filter

e.g. if we pass in `submissionId` right now, it won't filter by `uid`

Reply:
Ah! good catch. my bad.